include/dlaf/eigensolver/backtransformation/mc.h

```diff
@@ -0,0 +1,204 @@
+}
+
+template <class Executor, Device device, class T>
+void gemmUpdateW2(Executor&& ex, hpx::future<matrix::Tile<T, device>> w,
+                  hpx::shared_future<matrix::Tile<const T, device>> c,
+                  hpx::future<matrix::Tile<T, device>> w2) {
+  hpx::dataflow(ex, matrix::unwrapExtendTiles(tile::gemm_o), blas::Op::ConjTrans, blas::Op::NoTrans,
+                T(1.0), w, c, T(1.0), std::move(w2));
+}
+
+template <class Executor, Device device, class T>
+void gemmTrailingMatrix(Executor&& ex, hpx::shared_future<matrix::Tile<const T, device>> v,
+                        hpx::shared_future<matrix::Tile<const T, device>> w2,
+                        hpx::future<matrix::Tile<T, device>> c) {
+  hpx::dataflow(ex, matrix::unwrapExtendTiles(tile::gemm_o), blas::Op::NoTrans, blas::Op::NoTrans,
+                T(-1.0), v, w2, T(1.0), std::move(c));
+}
+
+// Implementation based on:
+// 1. Algorithm 6 "LAPACK Algorithm for the eigenvector back-transformation", page 15, PhD thesis
+// "GPU Accelerated Implementations of a Generalized Eigenvalue Solver for Hermitian Matrices with
+// Systematic Energy and Time to Solution Analysis" presented by Raffaele Solcà (2016)
+// 2. G. H. Golub and C. F. Van Loan, Matrix Computations, chapter 5, The Johns Hopkins University Press
+template <class T>
+struct BackTransformation<Backend::MC, Device::CPU, T> {
+  static void call_FC(Matrix<T, Device::CPU>& mat_c, Matrix<const T, Device::CPU>& mat_v,
+                      common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus);
+};
+
+template <class T>
+void BackTransformation<Backend::MC, Device::CPU, T>::call_FC(
+    Matrix<T, Device::CPU>& mat_c, Matrix<const T, Device::CPU>& mat_v,
+    common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus) {
+  using hpx::unwrapping;
+
+  auto executor_hp = dlaf::getHpExecutor<Backend::MC>();
+  auto executor_np = dlaf::getNpExecutor<Backend::MC>();
+
+  const SizeType m = mat_c.nrTiles().rows();
+  const SizeType n = mat_c.nrTiles().cols();
+  const SizeType mb = mat_v.blockSize().rows();
+
+  if (m <= 1 || n == 0)
+    return;
+
+  const SizeType nr_reflector = mat_v.size().rows() - mb;
+
+  dlaf::matrix::Distribution dist_t({mb, nr_reflector}, {mb, mb});
+
+  constexpr std::size_t n_workspaces = 2;
+  common::RoundRobin<matrix::Panel<Coord::Col, T, Device::CPU>> panelsV(n_workspaces,
+                                                                        mat_v.distribution());
+  common::RoundRobin<matrix::Panel<Coord::Col, T, Device::CPU>> panelsW(n_workspaces,
+                                                                        mat_v.distribution());
+  common::RoundRobin<matrix::Panel<Coord::Row, T, Device::CPU>> panelsW2(n_workspaces,
+                                                                         mat_c.distribution());
+  common::RoundRobin<matrix::Panel<Coord::Row, T, Device::CPU>> panelsT(n_workspaces, dist_t);

```

Comment:
Still not clear why 2 panels for T are needed.

Reply:
It is not clear to me why "2 panel for T", sorry.  So there is a panel for `T` with 2 workspaces (or do you mean that I need only one of those?). And there is a panel for `W2`: I cannot re-use `panelT` because it has a different distribution. 
Thanks!

Reply:
You are creating a RoundRobin which contains two (n_workspaces) panels.
As for T only a Tile per iteration is needed, a single panel is more than enough. (And no need for the RoundRobin in this case...)

Reply:
now I got it, thanks! is that ok?